### PR TITLE
Trace fix for the new pipeline

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
- With the migration to new pipeline we are not creating Aggs struct member of the response. Due to this 0 results were being shown for traces even though traces were present.
- Added code to use default fields of the response.
- We should ensure in future that other parts of the codebase should only use default fields, extra fields present in the response is only meant for the client and not internal code.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?
- Ingest the data using the [GoApp](https://www.siglens.com/siglens-docs/instrument-traces/go-app)
- Ingested different traces GET, POST.
- Queried the traces and verified that the traces returned are correct.

image

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/tracing/handler/tracehandler.go [hunk 1]
```diff
@@ -207,9 +208,35 @@ func ParseAndValidateRequestBody(ctx *fasthttp.RequestCtx) (*structs.SearchReque
...
+	if config.IsNewQueryPipelineEnabled() {
+		return pipeSearchResponseOuter.BucketCount
+	}
 	return len(pipeSearchResponseOuter.Aggs[""].Buckets)
 }
+
 func GetUniqueTraceIds(pipeSearchResponseOuter *segstructs.PipeSearchResponseOuter, startEpoch uint64, endEpoch uint64, page int) []string {
+	if config.IsNewQueryPipelineEnabled() {
+		totalTracesIds := GetTotalUniqueTraceIds(pipeSearchResponseOuter)
+		if totalTracesIds < (page-1)*50 {
```
Comment (nkunal) [inline pkg/segment/tracing/handler/tracehandler.go:220]: make this 50 a const

File: pkg/segment/tracing/handler/tracehandler.go [hunk 2]
```diff
@@ -207,9 +208,35 @@ func ParseAndValidateRequestBody(ctx *fasthttp.RequestCtx) (*structs.SearchReque
...
+		}
+
+		endIndex := page * 50
+		if endIndex > totalTracesIds {
+			endIndex = totalTracesIds
+		}
+
+		traceIds := make([]string, 0)
+		// Only Process up to 50 traces per page
+		for _, bucket := range pipeSearchResponseOuter.MeasureResults[(page-1)*50 : endIndex] {
```
Comment (nkunal) [inline pkg/segment/tracing/handler/tracehandler.go:231]: same comment here

File: pkg/segment/tracing/handler/tracehandler.go (hunk 1 above)
Comment (vasukalariya) [reply pkg/segment/tracing/handler/tracehandler.go]: addressed in the latest commit.

File: pkg/segment/tracing/handler/tracehandler.go (hunk 2 above)
Comment (vasukalariya) [reply pkg/segment/tracing/handler/tracehandler.go]: addressed in the latest commit.

**Review by nkunal — APPROVED** [review]

**Pull request merged**